### PR TITLE
Support inline flag

### PR DESCRIPTION
**Pull request by morlay**
for `additionalProperties` https://github.com/golang/go/issues/6213 as https://github.com/go-yaml/yaml did

### CODE REVIEW

**Comment by Quentin-M** [comment]
+1

**Comment by Quentin-M** [comment]
Looks broken tho?

```
//easyjson:json
type Spec struct {
	Version    string          `json:"version" yaml:"version"`
	Components []ComponentSpec `json:"components" yaml:"components"`
	Parameters interface{}     `json:"parameters" yaml:"parameters"`
}

type ComponentSpec struct {
	Kind   string                 `json:"kind" yaml:"kind"`
	Params map[string]interface{} `json:",inline" yaml:",inline"`
}

func main() {
	b := `{"version": "v1", "components": [{"kind": "testResource", "template": "a.yaml"}, {"kind": "testResource", "template": "b.yaml"}]}`
	s, err := NewSpec([]byte(b))
	fmt.Printf("%#v\n", s)
	fmt.Println(err)
}

func NewSpec(b []byte) (*Spec, error) {
	spec := &Spec{}

	// Unmarshal.
	err := spec.UnmarshalJSON(b)
	if err != nil {
		return nil, err
	}
        return spec, nil
}
```

```
&main.Spec{
  Version:"v1", 
  Components:[]main.ComponentSpec{
    main.ComponentSpec{
      Kind:"testResource",
      Params:map[string]interface {}{
        "version":"v1", 
        "components":[]interface {}{
          map[string]interface {}{
            "kind":"testResource", "template":"a.yaml"
          },
          map[string]interface {}{
            "kind":"testResource", "template":"b.yaml"
          }
        }
      }
    },
    main.ComponentSpec{
      Kind:"testResource", 
      Params:map[string]interface {}{
        "version":"v1", 
        "components":[]interface {}{
          map[string]interface {}{
            "kind":"testResource","template":"a.yaml"
          },
          map[string]interface {}{
            "kind":"testResource", "template":"b.yaml"
          }
        }
      }
    }
  }, 
  Parameters:interface {}(nil)
}
```

File: gen/decoder.go [hunk 1]
```diff
@@ -257,14 +257,31 @@ func (g *Generator) genTypeDecoderNoCheck(t reflect.Type, out string, tags field
...
 
 	if tags.omit {
 		return nil
 	}
 
+	if inline {
+		if tags.inline {
+			// each inline should clone Lexer Data
+			fmt.Fprintln(g.out, "{")
+			fmt.Fprintln(g.out, "  in := &jlexer.Lexer{Data: in.Data}")
```
Comment (Quentin-M) [inline gen/decoder.go]: You are making a fresh lexer with all the data, but without the position. Thus, you are reading from the beginning of the entire JSON document.

**Comment by rvasily** [comment]
We trying to keep maxumum compartability with stdlib, so i wont merge this before it appears in stdlib enconding/json

**Pull request closed without merging**